src/chainstate/stacks/db/accounts.rs

```diff
@@ -555,178 +521,143 @@ impl StacksChainState {
-        // * R is the number of tokens added by all blocks mined in this interval
-        // * b is the amount of burn tokens destroyed by this miner
-        // * B is the total amount of burn tokens destroyed over this interval
-        let coinbase_reward: u128 = if num_mined == 0 {
-            // this miner didn't help at all
-            0
+        let burn_total = other_burn_total
+            .checked_add(this_burn_total)
+            .expect("FATAL: combined burns exceed u128");
+
+        test_debug!(
+            "{}: Coinbase reward = {} * ({}/{})",
+            participant.address.to_string(),
+            participant.coinbase,
+            this_burn_total,
+            burn_total
+        );
+
+        // each participant gets a share of the coinbase proportional to the fraction it burned out
+        // of all participants' burns.
+        let coinbase_reward = participant
+            .coinbase
+            .checked_mul(this_burn_total as u128)
+            .expect("FATAL: STX coinbase reward overflow")
+            / (burn_total as u128);
+
+        // process poison -- someone can steal a fraction of the total coinbase if they can present
+        // evidence that the miner forked the microblock stream.  The remainder of the coinbase is
+        // destroyed if this happens.
+        let (recipient, coinbase_reward) = if let Some(reporter_address) = poison_reporter_opt {
+            if participant.address == miner.address {
+                // the poison-reporter, not the miner, gets a (fraction of the) reward
+                debug!(
+                    "{:?} will recieve poison-microblock commission {}",
+                    &reporter_address.to_string(),
+                    StacksChainState::poison_microblock_commission(coinbase_reward)
+                );
+                (
+                    reporter_address.clone(),
+                    StacksChainState::poison_microblock_commission(coinbase_reward),
+                )
+            } else {
+                // users that helped a miner that reported a poison-microblock get nothing
+                (StacksAddress::burn_address(mainnet), coinbase_reward)
+            }
         } else {
-            let burn_total = other_burn_total
-                .checked_add(this_burn_total)
-                .expect("FATAL: combined burns exceed u128");
-            test_debug!(
-                "{}: Coinbase reward = {} * ({}/{})",
-                miner.address.to_string(),
-                miner.coinbase,
-                this_burn_total,
-                burn_total
-            );
-            miner
-                .coinbase
-                .checked_mul(this_burn_total as u128)
-                .expect("FATAL: STX coinbase reward overflow")
-                / (burn_total as u128)
+            // no poison microblock reported
+            (participant.address, coinbase_reward)
         };
 
         // TODO: missing transaction fee calculation
```

Comment:
We might be able to mark this todo as resolved

Reply:
It's not resolved -- we're not yet including the transaction fees, nor are we calculating the 40/60 microblock stream split (which is waiting on a transaction fee calculation model).